cmd/ocitools/validate.go

```diff
@@ -381,6 +386,47 @@ func checkLinux(spec rspec.Spec, rootfs string, hostCheck bool) (msgs []string)
 	return
 }
 
+func checkLinuxResources(r rspec.Resources, hostCheck bool) (msgs []string) {
+	logrus.Debugf("check linux resources")
+
+	if r.OOMScoreAdj != nil && (int(*r.OOMScoreAdj) < -1000 || int(*r.OOMScoreAdj) > 1000) {
+		msgs = append(msgs, fmt.Sprintf("OOMScoreAdj range is from -1000 to 1000"))
```

Comment:
On Wed, Aug 17, 2016 at 06:28:27PM -0700, Ma Shimiao wrote:
> I think ocitools should have this functions when validating bundle.
> can we merge this into ocitools?

I don't think we need to merge the tool, since [1] is so tiny.  My
preferred approach would be to have something like
opencontainers/runtime-spec#490 so we could pull the schema JSON from
opencontainers.org (if folks ever get around to publishing it there ;)
or raw.githubusercontent.com (if they don't).  The ocitools validate
flow would be:

1. Invoked on a given config.
2. Find the ociVersion.
3. Figure out the URL for the associated JSON Schema (e.g. [2]).
4. Have xeipuuv/gojsonschema or similar use that schema to validate
   the config.
5. Use Go (or Python #98) to perform the checks that JSON Schema can't
   handle (e.g. multi-param checks, host-specific checks, …).

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/schema/validate.go
[2]: https://raw.githubusercontent.com/opencontainers/runtime-spec/v1.0.0-rc1/schema/schema.json